src/pages/Admin/Admin.jsx

```diff
@@ -97,63 +129,79 @@ const Admin = () => {
-              : <DeleteIcon
-                  color="error"
-                  sx={{ cursor: "pointer" }}
-                  onClick={() => deleteTitleData(dataList._id)}
-                />
-              
-            }
-            InnerOption={dataList.options.map((option, index) => (
-              <InnerPoll
-                option={option.option}
-                key={index}
-                votes={option.vote}
-                deleteOption={
-                 
-                  optionData === option.option &&
-                  deleteOptionLoading ?
+              }
+              AddTitle={
+                dataList.options.length < 4 && (
+                  <Link
+                    to={`AddOption/${dataList._id}`}
+                    style={{ color: "black" }}
+                    state={dataList.title}
+                  >
+                    <AddIcon />
+                  </Link>
+                )
+              }
+              deleteTitle={
+                dataList._id === deleteTitleId && deleteTitleLoading ? (
                   <CircularProgress color="inherit" />
-                  : <DeleteIcon
-                      color="error"
-                      sx={{ cursor: "pointer" }}
-                      onClick={() =>
-                        deleteOptionData(dataList._id, option.option)
-                      }
-                    />
-                }
-              />
-            ))}
-          />
-        ))}
+                ) : (
+                  <DeleteIcon
+                    color="error"
+                    sx={{ cursor: "pointer" }}
+                    onClick={() => deleteTitleData(dataList._id)}
+                  />
+                )
+              }
+              InnerOption={dataList.options.map((option, index) => (
+                <InnerPoll
+                  option={option.option}
+                  key={index}
+                  votes={option.vote}
+                  deleteOption={
+                    optionData === option && deleteOptionLoading ? (
+                      <CircularProgress color="inherit" />
+                    ) : (
+                      <DeleteIcon
+                        color="error"
+                        sx={{ cursor: "pointer" }}
+                        onClick={() =>
+                          deleteOptionData(
+                            dataList._id,
+                            option,dataList.options
+                          )
+                        }
+                      />
+                    )
+                  }
+                />
+              ))}
+            />
+          ))}
       </Box>
+      <TablePagination
+        component="div"
+        rowsPerPageOptions={rowsPerPageOptions}
+        count={pollList.data.length}
+        page={
+          page >= Math.ceil(pollList.data.length / rowPerPage)
+            ? Math.max(0, Math.ceil(pollList.data.length / rowPerPage) - 1)
+            : page
```

Comment:
Can you separate this logic to a function that return actual value after calculations,
use that function here